packages/cursorless-vscode/src/keyboard/KeyboardCommandsTargeted.ts

```diff
@@ -175,59 +176,80 @@ export default class KeyboardCommandsTargeted {
-        case "replaceWithTarget":
-        case "moveToTarget":
-          return {
-            name,
-            source: target,
-            destination: { type: "implicit" },
-          };
-        case "swapTargets":
-          return {
-            name,
-            target1: target,
-            target2: { type: "implicit" },
-          };
-        case "callAsFunction":
-          return {
-            name,
-            callee: target,
-            argument: { type: "implicit" },
-          };
-        case "pasteFromClipboard":
-          return {
-            name,
-            destination: {
-              type: "primitive",
-              insertionMode: "to",
+  performSimpleActionOnTarget = async (
+    actionDescription: KeyboardActionDescriptor,
+  ) => {
+    let name = "";
+    let exitCursorlessMode = false;
+    if (isString(actionDescription)) {
+      name = actionDescription;
+    } else {
+      name = actionDescription.actionId;
+      exitCursorlessMode = actionDescription.exitCursorlessMode ?? false;
+    }
+
+    return this.performActionOnTarget(
+      (target: PartialPrimitiveTargetDescriptor) => {
+        let action: ActionDescriptor;
+        switch (name) {
+          case "wrapWithPairedDelimiter":
+          case "rewrapWithPairedDelimiter":
+          case "insertSnippet":
+          case "wrapWithSnippet":
+          case "executeCommand":
+          case "replace":
+          case "editNew":
+          case "getText":
+            throw Error(`Unsupported keyboard action: ${name}`);
+          case "replaceWithTarget":
+          case "moveToTarget":
+            action = {
+              name,
+              source: target,
+              destination: { type: "implicit" },
+            };
+            break;
+          case "swapTargets":
+            action = {
+              name,
+              target1: target,
+              target2: { type: "implicit" },
+            };
+            break;
+          case "callAsFunction":
+            action = {
+              name,
+              callee: target,
+              argument: { type: "implicit" },
+            };
+            break;
+          case "pasteFromClipboard":
+            action = {
+              name,
+              destination: {
+                type: "primitive",
+                insertionMode: "to",
+                target,
+              },
+            };
+            break;
+          case "generateSnippet":
+          case "highlight":
+            action = {
+              name,
               target,
-            },
-          };
-        case "generateSnippet":
-        case "highlight":
-          return {
-            name,
-            target,
-          };
-        default:
-          return {
-            name,
-            target,
-          };
-      }
-    });
+            };
+            break;
+          default:
+            action = {
+              name: name as any,
```

Comment:
Any time you're forced to cast to `any`, it's a bad smell. We used to be a bit more fast and loose with our typing, and we kept getting bugs where we could trace it to this kinda thing. There's always a trade-off whether it's worth it to keep fighting with the types or give up, and I don't think the answer is always to fight till the end and have perfect strong types, but we def err more on the side of getting the right types after enough bugs that were caused by subverting the type system